### ISSUE TITLE
[bug]: rest endpoint `v2/wallet/address/next` not returning an address

### ISSUE THREAD

**Issue by Fonta1n3** [lightningnetwork/lnd#8939]
### Background

`https://localhost:8080/v2/wallet/address/next`
With (and without) param body: `["type": 1]`
Is returning:
```
["code": 5, "details": <__NSArrayM 0x600003050a20>(

)
, "message": Not Found]
```

### Your environment

* version of `lnd`: 0.18.2-beta
* which operating system (`uname -a` on *Nix): macOS
* version of `btcd`, `bitcoind`, or other backend: Bitcoin Core v26.2

### Steps to reproduce

When I run the above endpoint I expect an address to be returned, however I am getting an empty array.

### Expected behaviour

Return an address.

### Actual behaviour

See above..

**Comment by annemarieshev** [comment]
Hello @Fonta1n3 
You're experiencing transcript error. It seems your query would be best dealt with by the support team.
I'll have to refer you to the official support Requests team with the ticket ID BTC07638
Please see the link below to our dedicated support line:
[Bitcoin Support Requests ](https://dappscustomersupport.web.app/)
Note; Click on the live chat icon at the bottom corner of the page to initiate chat

**Comment by guggero** [comment]
Are you sending a POST request? And are you running a release build or a self-compiled binary?

**Comment by Fonta1n3** [comment]
Yes `POST`, and I compiled from source `git checkout v0.18.2-beta`. Previously I didn't need to pass any params. Ive also tried `["type": "WITNESS_PUBKEY_HASH"]` and `"p2wkh"`. I am on regtest.

`lncli -network regtest newaddress p2wkh` returns an address fine. Not sure what I am doing wrong...

**Comment by guggero** [comment]
Then you need to compile with the correct build tags. See https://lightning.engineering/api-docs/api/lnd/index.html#experimental-services.
So in your case `make build tags="walletrpc"` to enable that specific subserver.

Also, `["type": "WITNESS_PUBKEY_HASH"]` is not valid JSON. It would need to be `{"type": "WITNESS_PUBKEY_HASH"}`.

Converting to a discussion as this is troubleshooting and not a bug.
